### PR TITLE
DVPN 136 - Schedule ETL for cjms prod

### PR DESCRIPTION
**Pull request by relud**

### CODE REVIEW

File: sql/moz-fx-cjms-nonprod-9a36/cjms_bigquery/subscriptions_v1/query.sql [hunk 1]
```diff
@@ -38,10 +38,10 @@ attributed_flows AS (
     subscription_created
 ),
 attributed_subs AS (
-  -- last subscription for each flow, for 1:1 relationship between flow and subscription
+  -- first subscription for each flow, for 1:1 relationship between flow and subscription
   SELECT
     flow_id,
-    ARRAY_AGG(subscription_id ORDER BY subscription_created DESC LIMIT 1)[
+    ARRAY_AGG(subscription_id ORDER BY subscription_created LIMIT 1)[
```
Comment (relud) [inline sql/moz-fx-cjms-nonprod-9a36/cjms_bigquery/subscriptions_v1/query.sql:44]: per https://github.com/mozilla/bigquery-etl/pull/2823#discussion_r835615236 I discussed this with product, and we do indeed want first subscription by flow.

File: bigquery_etl/query_scheduling/templates/airflow_dag.j2 [hunk 2]
```diff
@@ -151,6 +152,7 @@ with DAG('{{ name }}', default_args=default_args{%+ if schedule_interval != None
 
     {% if task.depends_on_fivetran != None -%}
     {% for fivetran_task in task.depends_on_fivetran -%}
+    {% if fivetran_task not in fivetran_seen -%}
```
Comment (relud) [inline bigquery_etl/query_scheduling/templates/airflow_dag.j2:155]: fixes bug where fivetran_tasks would be defined mulitple times if referenced multiple times, by following the same pattern used for external dependencies

File: sql/moz-fx-cjms-prod-f3c7/cjms_bigquery/subscriptions_v1/query.sql [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  SELECT
+    stripe_subscriptions.subscription_id,
+    stripe_subscriptions.created AS subscription_created,
+    ARRAY_AGG(aic_flows.flow_id ORDER BY aic_flows.flow_started DESC LIMIT 1)[
+      SAFE_OFFSET(0)
+    ] AS flow_id,
+  FROM
+    aic_flows
+  JOIN
+    mozdata.subscription_platform.stripe_subscriptions
```
Comment (relud) [inline sql/moz-fx-cjms-prod-f3c7/cjms_bigquery/subscriptions_v1/query.sql:32]: I discussed with product, and we do not want to limit what products can be attributed during the MVP.

File: sql/moz-fx-cjms-prod-f3c7/cjms_bigquery/subscriptions_v1/query.sql [hunk 4]
```diff
@@ -0,0 +1,68 @@
+WITH aic_flows AS (
+  -- last fxa_uid for each flow_id
+  SELECT
+    flow_id,
+    MIN(flow_started) AS flow_started,
+    ARRAY_AGG(fxa_uid IGNORE NULLS ORDER BY fxa_uid_timestamp DESC LIMIT 1)[
+      SAFE_OFFSET(0)
+    ] AS fxa_uid,
```
Comment (relud) [inline sql/moz-fx-cjms-prod-f3c7/cjms_bigquery/subscriptions_v1/query.sql:8]: I discussed with product, and we don't care about edge cases with earlier `fxa_uid`s during the MVP

File: sql/moz-fx-cjms-prod-f3c7/cjms_bigquery/subscriptions_v1/query.sql [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  FROM
+    `moz-fx-cjms-prod-f3c7`.cjms_bigquery.flows_v1
+  JOIN
+    EXTERNAL_QUERY("moz-fx-cjms-prod-f3c7.us.cjms-sql", "SELECT flow_id FROM aic")
+  USING
+    (flow_id)
+  WHERE
+    -- only use the last 10 days in stage
+    submission_date >= CURRENT_DATE - 10
```
Comment (sean-rose) [inline sql/moz-fx-cjms-prod-f3c7/cjms_bigquery/subscriptions_v1/query.sql:17]: Should this be different for production?

**Review by sean-rose — APPROVED** [review]

File: sql/moz-fx-cjms-prod-f3c7/cjms_bigquery/subscriptions_v1/query.sql (hunk 5 above)
Comment (relud) [inline sql/moz-fx-cjms-prod-f3c7/cjms_bigquery/subscriptions_v1/query.sql:17]:
```suggestion
    -- use the last 31 days in prod
    submission_date >= CURRENT_DATE - 31
```

Comment (relud) [reply sql/moz-fx-cjms-prod-f3c7/cjms_bigquery/subscriptions_v1/query.sql:17]: yes, updating to 31 days

**Pull request merged**